warp10/src/main/java/io/warp10/script/mapper/FILLMAPPER.java

```diff
@@ -0,0 +1,90 @@
+//
+//   Copyright 2020  SenX S.A.S.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.script.mapper;
+
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptException;
+import io.warp10.script.WarpScriptLib;
+import io.warp10.script.WarpScriptMapperFunction;
+import io.warp10.script.WarpScriptStack;
+import io.warp10.script.WarpScriptStackFunction;
+
+/**
+ * Wrap a Mapper so empty buckets are filled by the mapper and non-empty buckets are left untouched
+ */
+public class FILLMAPPER extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+  public FILLMAPPER(String name) {
+    super(name);
+  }
+
+  public static final class FillerMapper extends NamedWarpScriptFunction implements WarpScriptMapperFunction {
+
+    private final WarpScriptMapperFunction mapper;
+
+    public FillerMapper(String name, WarpScriptMapperFunction mapper) {
+      super(name);
+      this.mapper = mapper;
+    }
+
+    @Override
+    public Object apply(Object[] args) throws WarpScriptException {
+
+      long[] ticks = (long[]) args[3];
+      long tick = (long) args[0];
+      int tick_index = (int) ((long[]) args[7])[4];
```

Comment:
Explain what this is, args[7][4] is kind of cryptic for someone not knowing what args contains.